src/lib/AclPermission.ts

```diff
@@ -21,15 +21,29 @@ export class AclPermission {
         this.attributes = params.attributes;
     }
     public filter(data: any): any {
-        const result = this.attributes
+        let result = this.attributes
+            .filter(attribute => attribute.charAt(0) !== '!')
             .map(attribute => jp.paths(data, `$..${attribute}`))
             .reduce((a, b) => a.concat(b), [])
-            .map((jpPath: string[]) => jpPath.slice(1))
-            .map((jpPath: string[]) => [jpPath, path(jpPath, data)])
-            .reduce((acc: any, [jpPath, data]: [string[], any]) => assocPath(jpPath, data, acc), {});
+            .map(jpPath => jpPath.slice(1))
+            .map(jpPath => [jpPath, path(jpPath, data)])
+            // @ts-ignore
```

Comment:
Please, specify ignored rule

Reply:
The `jpPath` is `unknown` - I added the `as string[]`